src/Diagnostics.RuntimeHost/Services/StorageService/StorageService.cs

```diff
@@ -126,6 +140,49 @@ public async Task<DiagEntity> LoadDataToTable(DiagEntity detectorEntity)
             }
         }
 
+        public async Task<string> LoadBlobToContainer(string blobname, string contents)
+        {
+            try
+            {
+                var timeTakenStopWatch = new Stopwatch();
+                await containerClient.CreateIfNotExistsAsync();       
+                timeTakenStopWatch.Start();
+                var cloudBlob = containerClient.GetBlockBlobReference(blobname);
+                using (var uploadStream = new MemoryStream(Convert.FromBase64String(contents)))
+                {
+                    await cloudBlob.UploadFromStreamAsync(uploadStream);               
+                }
+                await cloudBlob.FetchAttributesAsync();
+                timeTakenStopWatch.Stop();
+                var uploadResult = cloudBlob.Properties;  
+                DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageService), $"Loaded {blobname}, etag {uploadResult.ETag}, time taken {timeTakenStopWatch.ElapsedMilliseconds}");
+                return uploadResult.ETag;
+            } catch (Exception ex)
```

Comment:
Are you going to do any retries?

Reply:
The storage client library has built-in retry policies. https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#azure-storage

Reply:
Is it built in or do you have to configure it? When I read the docs you sent me it looks like you have to configure it with some `BlobRequestOptions` class

Reply:
So we can use `BlobRequestOptions` if we want to configure it manually. These are default values:

![image](https://user-images.githubusercontent.com/46545195/81125606-7e904980-8eed-11ea-9be2-b6b79767a838.png)